qemu/tests/rh_qemu_update.py

```diff
@@ -0,0 +1,309 @@
+            test.log.info("Installation of '{}' succeeded".format(
+                params['qemu_package_install']
+                )
+            )
+
+    def install_component_management(session):
+        for pkg_name in params["component_management_preqs"].split():
+            cmd = f"{params['pm_tool']} install -y {pkg_name}"
+            s, o = session.cmd_status_output(cmd, timeout=INSTALL_TIMEOUT)
+            if s != 0:
+                test.error(f"Installation of '{pkg_name}'' failed with: {o}")
+            else:
+                test.log.info(f"Installation of '{pkg_name}' succeeded")
+        cmd_clone = "git clone --depth=1 {} -b {}".format(
+            params['component_management_repo'],
+            params['component_management_branch']
+        )
+        s_clone, o_clone = session.cmd_status_output(
+            cmd_clone, timeout=OPERATION_TIMEOUT
+            )
+        if s_clone != 0:
+            test.error("Clonning of '{}' failed with: {}".format(
+                params['component_management_repo'], o_clone
+                )
+            )
+        else:
+            test.log.info("Clonning of '{}' succeeded".format(
+                params['component_management_repo']
+                )
+            )
+
+    def _compare_qemu_versions(nvr_1, nvr_2):
+        """
+        Compare two qemu vesrion based on their NVRs, function returns tuple
+        containing 3 items: bool (nvr_1 is newer than nvr_2) and both NVRs
+        """
+        def nvr_to_numbers_list(nvr):
+            nvr_list = re.split(
+                "[-.]", nvr.replace("qemu-kvm", "").split("module")[0][1:]
+                )
+            # Maximum expected number of digits in version is 4
+            # for example of 'qemu-kvm-6.2.0-11' is [6, 2, 0, 11]
+            return [int(i) for i in nvr_list if i.isnumeric()][:4]
+
+        def nice_version_format(ugly_list):
+            return ".".join(str(num) for num in ugly_list)
+        qemu_version_1 = nvr_to_numbers_list(nvr_1)
+        qemu_version_2 = nvr_to_numbers_list(nvr_2)
+        # If versions have different number of digits
+        # add 0 istead of keeping empty position
+        if len(qemu_version_1) > len(qemu_version_2):
+            qemu_version_2.append(0)
+        elif len(qemu_version_1) < len(qemu_version_2):
+            qemu_version_1.append(0)
+        if len(qemu_version_1) != len(qemu_version_2):
+            raise ValueError(f"One of NVRs is invalid: {nvr_1} {nvr_2}")
+        for i in range(len(qemu_version_1)):
+            if qemu_version_1[i] > qemu_version_2[i]:
+                return (
+                    True,
+                    nice_version_format(qemu_version_1),
+                    nice_version_format(qemu_version_2)
+                    )
+        return (
+            False,
+            nice_version_format(qemu_version_1),
+            nice_version_format(qemu_version_2)
+            )
+
+    def _get_installed_qemu_info(session, location="guest"):
```

Comment:
How do you think about use `session` to indicate it is a guest or host ?

Reply:
Good idea, it will spare us some complexity, which is not necessary. Fixed.